### PR TITLE
Master refactor

### PR DESCRIPTION
**Pull request by EbenezerGH**

### CODE REVIEW

File: core/src/main/kotlin/com/props/core/wallet/HDKeyBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.bitcoinj.wallet.DeterministicSeed
+import java.security.SecureRandom
+
+@OpenForTesting
+internal class HDKeyBuilder(internal var context: Context) : HDKeyBuilderI {
+
+    final var mnemonicCode: MnemonicCode =
+            MnemonicCode(this.context.resources.openRawResource(R.raw.mnemonic), null)
+
+    override fun generateHDKey(password: String, seedLen: MnemonicSeedLength): WalletFileBuilderI {
```
Comment (kc1116) [inline core/src/main/kotlin/com/props/core/wallet/HDKeyBuilder.kt]: add default password value, so were not relying on what the library uses

File: core/src/main/kotlin/com/props/core/wallet/WalletFileBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if (destinationDir.exists()) destinationDir = File(internalStoreFolder, "$walletName-${destinationDir.listFiles().size}")
+        destinationDir.mkdirs()
+
+        val executor = Executors.newSingleThreadExecutor()
+        val result = executor.submit(Callable<String> {
+            WalletUtils.generateWalletFile(password, credentials.ecKeyPair, destinationDir, false)
+        })
+
+        executor.shutdown()
+        while (!result.isDone) {
```
Comment (kc1116) [inline core/src/main/kotlin/com/props/core/wallet/WalletFileBuilder.kt]: if we are waiting before returning is this still blocking main thread ? Might want to return immediately and use some call back pattern

File: core/src/main/kotlin/com/props/core/backup/DriveBackupClient.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                            .build()
+
+                    val results = doQuery(driveResourceClient
+                            .queryChildren(getAppFolder(), query)).first().driveId
+
+                    parseBackupFile(results)
+                })
+
+                threadPool.shutdown()
+                return result.get()
```
Comment (bikranttripathi) [inline core/src/main/kotlin/com/props/core/backup/DriveBackupClient.kt]: If the app calls this from the main thread, will this block the thread until it finishes the future task?

Comment (EbenezerGH) [reply core/src/main/kotlin/com/props/core/backup/DriveBackupClient.kt]: right, I will call this differently

File: core/src/main/kotlin/com/props/core/wallet/WalletFileBuilder.kt (hunk 2 above)
Comment (EbenezerGH) [reply core/src/main/kotlin/com/props/core/wallet/WalletFileBuilder.kt]: got it

**Pull request closed without merging**